vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go

```diff
@@ -36,8 +36,8 @@ type recommendationProvider struct {
 	recommendationProcessor vpa_api_util.RecommendationProcessor
 }
 
-// NewRecommendationProvider constructs the recommendation provider that can be used to determine recommendations for pods.
-func NewRecommendationProvider(calculator limitrange.LimitRangeCalculator,
+// NewProvider constructs the recommendation provider that can be used to determine recommendations for pods.
+func NewProvider(calculator limitrange.LimitRangeCalculator,
 	recommendationProcessor vpa_api_util.RecommendationProcessor) *recommendationProvider {
```

Comment:
nit: I know it unrelated to this PR, but shouldn't this function return interface instead of a pointer?

Reply:
I'm not going to block the PR because of the unrelated nit. Other things look good.